include/pagmo/problems/lennard_jones.hpp

```diff
@@ -0,0 +1,182 @@
+#define PAGMO_PROBLEM_LENNARDJONES_HPP
+
+#include <iostream>
+#include <limits>
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <pagmo/exceptions.hpp>
+#include <pagmo/problem.hpp> // needed for cereal registration macro
+#include <pagmo/types.hpp>
+
+namespace pagmo
+{
+
+/// The Lennard Jones Cluster problem.
+/**
+ *
+ * \image html lennard_jones.jpg "Pictures of Lennard-Jones clusters." width=3cm
+ *
+ * This is a box-constrained continuous single-objecive problem. It represents the minimization
+ * of the energy of a cluster of atoms assuming a Lennard-Jones potential between each pair.
+ * The complexity for computing the objective function scales with the square of the number of atoms.
+ *
+ * The decision vector contains [z2, y3, z3, x4, y4, z4, ....] as the cartesian coordinates x1, y1, z1, x2, y2 and x3
+ * are fixed to be zero.
+ *
+ * See: http://doye.chem.ox.ac.uk/jon/structures/LJ.html
+ *
+ */
+class lennard_jones
+{
+public:
+    /// Constructor from number of atoms
+    /**
+     * Constructs a Lennard Jones Clusters global optimisation problem
+     *
+     * @param atoms number of atoms in the cluster.
+     *
+     * @throw std::invalid_argument if \p atoms is < 3
+     */
+    lennard_jones(unsigned atoms = 3u) : m_atoms(atoms)
+    {
+        if (atoms < 3) {
+            pagmo_throw(std::invalid_argument, "The number of atoms in a Lennard Jones Clusters problem must be "
+                                               "positive and greater than 2, while a number of "
+                                                   + std::to_string(atoms) + " was detected.");
+        }
+    }
+    /// Fitness computation
+    /**
+     * Computes the fitness for this UDP
+     *
+     * @param x the decision vector.
+     *
+     * @return the fitness of \p x.
+     */
+    vector_double fitness(const vector_double &x) const
+    {
+        vector_double f(1, 0.);
+        double sixth, dist;
+        f[0] = 0;
+        // We evaluate the potential
+        for (unsigned i = 0u; i < (m_atoms - 1u); ++i) {
+            for (unsigned j = (i + 1u); j < m_atoms; ++j) {
+                dist = std::pow(_r(i, 0u, x) - _r(j, 0u, x), 2) + std::pow(_r(i, 1u, x) - _r(j, 1u, x), 2)
+                       + std::pow(_r(i, 2u, x) - _r(j, 2u, x), 2); // rij^2
+                if (dist == 0.0) {
+                    f[0] = std::numeric_limits<double>::max();
+                } else {
+                    sixth = std::pow(dist, -3); // rij^-6
+                    f[0] += (std::pow(sixth, 2) - sixth);
+                }
+            }
+        }
+        f[0] = 4 * f[0];
+        return f;
+    }
+    /// Box-bounds
+    /**
+     * Returns the box-bounds for this UDP.
+     *
+     * @return the lower and upper bounds for each of the decision vector components
+     */
+    std::pair<vector_double, vector_double> get_bounds() const
+    {
+        unsigned prob_dim = 3u * m_atoms - 6u;
```

Comment:
There should an overflow check here:

```c++
if (m_atoms > std::numeric_limits<unsigned>::max() / 3u) {
    ...
}
```

However there's the same issue here as above, in mixing ``unsigned`` and ``vector_double::size_type``. So either we need 2 overflow checks, or we need to use consistently ``vector_double::size_type`` everywhere.